sampler.go

```diff
@@ -174,6 +181,141 @@ func (s *rateLimitingSampler) Equal(other Sampler) bool {
 
 // -----------------------
 
+type guaranteedThroughputProbabilisticSampler struct {
+	probabilisticSampler Sampler
```

Comment:
although I agree that I can strongly type these, in order to do so I'll either have to cast it after calling NewProbabilisticSampler since it returns a Sampler type or I'll have to change the New functions return signature and use duck typing. I don't think this is necessary since I don't have to access anything specific to the samplers (for now)